### PR TITLE
Change the slug for the theme.json doc to avoid conflicts

### PR DESCRIPTION
**Pull request by ndiego**
It turns out that each doc slug needs to be unique. A new theme.json doc was created in https://github.com/WordPress/gutenberg/pull/57289 and the slug was simply `theme-json`, which already existed. Therefore, the new doc was not synced to the Handbook. 

~~This PR modifies the slug to  `theme-json-configurations`, thus making it unique.~~

**Edit:** Following the [comment](https://github.com/WordPress/gutenberg/pull/57410#issuecomment-1870834215) below, I have updated the approach in this PR to instead change the slug of the [Global Settings & Styles](https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-json/) doc. While updating that doc, I also removed an unnecessary TOC.

> [!Warning]
> I have also created the redirect PR here: https://github.com/WordPress/wporg-developer/pull/457. This PR should be merged at the same time the redirect PR is put in place.

### CODE REVIEW

**Comment by t-hamano** [comment]
I feel it's better if the slug is based on the page title. The title of the page added in #57289 is "Global Settings & Styles (theme.json)", but the page title and slug are `theme-json`. How about changing this to `global-settings-and-styles` to avoid duplication?

In other words, the classification is as follows.

| Page Title | How-to Guides > Themes > Global Settings & Styles (theme.json)| How-to Guides > Curating the Editor Experience > theme.json |
|--------|--------|--------|
| Slug | `global-settings-and-styles` | `theme.json` | 
| Source | `../docs/how-to-guides/themes/global-settings-and-styles.md` | `../docs/how-to-guides/curating-the-editor-experience/theme-json.md` |

Reference: [Similar page in the theme handbook](https://developer.wordpress.org/themes/global-settings-and-styles/)

**Comment by ndiego** [comment]
> I feel it's better if the slug is based on the page title. The title of the page added in https://github.com/WordPress/gutenberg/pull/57289 is "Global Settings & Styles (theme.json)", but the page title and slug are theme-json. How about changing this to global-settings-and-styles to avoid duplication?

Yeah, I do like that better. Good call @t-hamano. This will require a new redirect to be put in place. I will get this PR and the redirect PR put together today.

**Comment by ndiego** [comment]
@t-hamano I have updated this PR based on your suggested approach. I have also created the redirect PR here: https://github.com/WordPress/wporg-developer/pull/457. This PR should be merged at the same time the redirect PR is put in place.

File: docs/how-to-guides/themes/theme-support.md [hunk 1]
```diff
@@ -315,7 +315,7 @@ Themes can opt out of generated block layout styles that provide default structu
 add_theme_support( 'disable-layout-styles' );
 ```
 
-For themes looking to customize `blockGap` styles or block spacing, see [the developer docs on Global Settings & Styles](/docs/how-to-guides/themes/theme-json/#what-is-blockgap-and-how-can-i-use-it).
+For themes looking to customize `blockGap` styles or block spacing, see [the developer docs on Global Settings & Styles](/docs/how-to-guides/themes/global-settings-and-styles/#what-is-blockgap-and-how-can-i-use-it).
```
Comment (t-hamano) [inline docs/how-to-guides/themes/theme-support.md]:
```suggestion
For themes looking to customize `blockGap` styles or block spacing, see [the developer docs on Global Settings & Styles](/docs/how-to-guides/themes/global-settings-and-styles.md#what-is-blockgap-and-how-can-i-use-it).
```

If the URL is a relative path starting with `/docs` and has an anchor, I think the link will not work properly on GitHub unless you include the markdown file extension.

https://github.com/WordPress/gutenberg/assets/54422211/79f291e8-5e72-40cd-9c97-8ab94b14f9b2

File: docs/reference-guides/block-api/block-supports.md [hunk 2]
```diff
@@ -437,7 +437,7 @@ _**Note:** Since WordPress 6.2._
 -   Subproperties:
     -   `minHeight`: type `boolean`, default value `false`
 
-This value signals that a block supports some of the CSS style properties related to dimensions. When it does, the block editor will show UI controls for the user to set their values if [the theme declares support](/docs/how-to-guides/themes/theme-json/#opt-in-into-ui-controls).
+This value signals that a block supports some of the CSS style properties related to dimensions. When it does, the block editor will show UI controls for the user to set their values if [the theme declares support](/docs/how-to-guides/themes/global-settings-and-styles/#opt-in-into-ui-controls).
```
Comment (t-hamano) [inline docs/reference-guides/block-api/block-supports.md]:
```suggestion
This value signals that a block supports some of the CSS style properties related to dimensions. When it does, the block editor will show UI controls for the user to set their values if [the theme declares support](/docs/how-to-guides/themes/global-settings-and-styles.md#opt-in-into-ui-controls).
```

File: docs/reference-guides/block-api/block-supports.md [hunk 3]
```diff
@@ -675,7 +675,7 @@ _**Note:** Since WordPress 6.2._
 -   Subproperties:
     -   `sticky`: type `boolean`, default value `false`
 
-This value signals that a block supports some of the CSS style properties related to position. When it does, the block editor will show UI controls for the user to set their values if [the theme declares support](/docs/how-to-guides/themes/theme-json/#opt-in-into-ui-controls).
+This value signals that a block supports some of the CSS style properties related to position. When it does, the block editor will show UI controls for the user to set their values if [the theme declares support](/docs/how-to-guides/themes/global-settings-and-styles/#opt-in-into-ui-controls).
```
Comment (t-hamano) [inline docs/reference-guides/block-api/block-supports.md]:
```suggestion
This value signals that a block supports some of the CSS style properties related to position. When it does, the block editor will show UI controls for the user to set their values if [the theme declares support](/docs/how-to-guides/themes/global-settings-and-styles.md#opt-in-into-ui-controls).
```

File: packages/block-editor/src/components/height-control/README.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 The `HeightControl` component adds a linked unit control and slider component for controlling the height of a block within the block editor. It supports passing a label, and is used for controlling the minimum height dimensions of Group blocks.
 
-_Note:_ It is worth noting that the minimum height option is an opt-in feature. Themes need to declare support for it before it'll be available, and a convenient way to do that is via opting in to the [appearanceTools](/docs/how-to-guides/themes/theme-json/#opt-in-into-ui-controls) UI controls.
+_Note:_ It is worth noting that the minimum height option is an opt-in feature. Themes need to declare support for it before it'll be available, and a convenient way to do that is via opting in to the [appearanceTools](/docs/how-to-guides/themes/global-settings-and-styles/#opt-in-into-ui-controls) UI controls.
```
Comment (t-hamano) [inline packages/block-editor/src/components/height-control/README.md]:
```suggestion
_Note:_ It is worth noting that the minimum height option is an opt-in feature. Themes need to declare support for it before it'll be available, and a convenient way to do that is via opting in to the [appearanceTools](/docs/how-to-guides/themes/global-settings-and-styles.md#opt-in-into-ui-controls) UI controls.
```

File: packages/style-engine/docs/using-the-style-engine-with-block-supports.md [hunk 5]
```diff
@@ -231,4 +231,4 @@ array(
 */
 ```
 
-Read more about [global styles](https://developer.wordpress.org/block-editor/explanations/architecture/styles/#global-styles) and [preset CSS custom properties](https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-json/#css-custom-properties-presets-custom) and [theme supports](https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-support/).
+Read more about [global styles](https://developer.wordpress.org/block-editor/explanations/architecture/styles/#global-styles) and [preset CSS custom properties](https://developer.wordpress.org/block-editor/how-to-guides/themes/global-settings-and-styles/#css-custom-properties-presets-custom) and [theme supports](https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-support/).
```
Comment (t-hamano) [inline packages/style-engine/docs/using-the-style-engine-with-block-supports.md]:
```suggestion
Read more about [global styles](https://developer.wordpress.org/block-editor/explanations/architecture/styles/#global-styles) and [preset CSS custom properties](https://developer.wordpress.org/block-editor/how-to-guides/themes/global-settings-and-styles.md#css-custom-properties-presets-custom) and [theme supports](https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-support/).
```

**Review by t-hamano — COMMENTED** [review]
I've made a few suggestions regarding links, and overall things seem fine 👍

In addition to this, I suspect that the link below is also not working properly.

https://github.com/WordPress/gutenberg/blob/4ca8767c7ad19bd932aab0064f9225cc6c27c5e5/docs/how-to-guides/themes/theme-support.md#L437

Perhaps `/docs/how-to-guides/curating-the-editor-experience/theme-json.md` rather than `./theme-json.md`?

File: docs/how-to-guides/themes/theme-support.md (hunk 1 above)
Comment (ndiego) [reply docs/how-to-guides/themes/theme-support.md]: Ahh, good catch.

**Comment by ndiego** [comment]
> Perhaps /docs/how-to-guides/curating-the-editor-experience/theme-json.md rather than ./theme-json.md?

Good catch, missed that one. All updated.

**Review by t-hamano — APPROVED** [review]
LGTM 🚀

**Pull request merged**